platform/migration/platform-migration/src/test/java/org/codice/ddf/configuration/admin/ConfigurationFilesPollerTest.java

```diff
@@ -86,362 +76,198 @@ public void setup() {
-                mock(Path.class));
-        when(mockConfigurationDirectory.register(mockWatchService,
-                StandardWatchEventKinds.ENTRY_CREATE)).thenThrow(new IOException());
-        ConfigurationFilesPoller configurationFilesPoller = new ConfigurationFilesPoller(
-                mockConfigurationDirectory,
-                FILE_EXT,
-                mockWatchService,
-                mockExecutorService);
-
-        // Perform Test
-        try {
-            configurationFilesPoller.init();
-        } catch (IOException e) {
-            // Verify
-            verify(mockExecutorService, never()).execute(configurationFilesPoller);
-            verify(mockWatchService, never()).take();
-            throw e;
-        }
-    }
-
     @Test
-    public void testResgiterFilesAlreadyExist() throws Exception {
+    public void testRunFileCreatedChangeListenerNotified() throws Exception {
         // Setup
-        Path mockPath = getMockPath(PID, FILE_EXT);
-        WatchEvent<?> mockWatchEvent = getMockWatchEvent(mockPath,
-                StandardWatchEventKinds.ENTRY_CREATE);
-        List<WatchEvent<?>> watchEvents = getSingleMockWatchEvent(mockWatchEvent);
-        WatchKey mockWatchKey = getMockWatchKey(watchEvents);
-        WatchService mockWatchService = getMockWatchService(mockWatchKey);
-        Path mockConfigurationDirectory = getMockConfigurationDirectoryPath(mockPath,
-                mock(Path.class));
-
-        Stream<Path> existingFiles = mock(Stream.class);
-        when(existingFiles.filter(any(Predicate.class))).thenReturn(Stream.of(configFilePath,
-                directoryPath,
-                nonConfigFilePath));
-        when(mockConfigurationDirectory.resolve(configFilePath)).thenReturn(configFilePath);
-
         ConfigurationFilesPoller configurationFilesPoller = new ConfigurationFilesPoller(
-                mockConfigurationDirectory,
-                FILE_EXT,
-                mockWatchService,
-                mockExecutorService) {
+                configFilePath,
+                FILE_EXT) {
             @Override
-            Stream<Path> getExistingFiles() throws IOException {
-                return existingFiles;
+            void waitForFileToBeCompletelyWritten(File file) throws InterruptedException {
+                return;
             }
         };
+        when(configFilePath.toAbsolutePath()).thenReturn(Paths.get("foo"));
+        File file = Files.createTempFile("foo", "bar")
+                .toFile();
 
         // Perform Test
         configurationFilesPoller.register(mockChangeListener);
+        configurationFilesPoller.onFileCreate(file);
 
         // Verify
-        verify(mockChangeListener).notify(configFilePath);
-        verify(mockChangeListener, never()).notify(directoryPath);
-        verify(mockChangeListener, never()).notify(nonConfigFilePath);
+        verify(mockChangeListener).notify(any());
```

Comment:
❗️ We should avoid using `any()` here and use a specific matcher instead, otherwise we're not validating that the listener is called with the proper file.